tools/system_libs.py

```diff
@@ -1376,6 +1376,10 @@ class libcxxabi(NoExceptLibrary, MTLibrary, DebugLibrary):
     ]
   includes = ['system/lib/libcxx/src']
 
+  def __init__(self, **kwargs):
+    super().__init__(**kwargs)
+    self.is_debug |= settings.EXCEPTION_STACK_TRACES
```

Comment:
Can also revert this I think?

Reply:
I don't think so? The point of this PR is to display stack traces when `ASSERTIONS` is not set but `EXXCEPTION_STACK_TRACES` is set, and 
```python
if settings.ASSERTIONS:
  default_setting('EXCEPTION_STACK_TRACES', 1)
```
This doesn't make `ASSERTIONS` set.

Reply:
I see, yes I agree then.  Can you add a comment here maybe"  "`EXCEPTION_STACK_TRACES` currently requires the debug version of libc++"    ...  BTW why is that?   Why do we need the debug version here?

Reply:
Because of this:
https://github.com/emscripten-core/emscripten/blob/6ac4b2a755543ffca5d57ced9130455a9d3ac1e4/system/lib/libcxxabi/src/cxa_exception.cpp#L290-L297

We could've come up with a separate preprocessor `define` keyword probably, but I thought this was simpler. Also enabling the debug version of libc++abi had its own advantages other than the stack traces, for example, it shows `abort` error messages, which the release version doesn't.

Reply:
Could we instead remove `__throw_exception_with_stack_trace` and have `_Unwind_RaiseException` modify its behaviour based on `EXCEPTION_STACK_TRACES` in JS code?  

Or is `_Unwind_RaiseException` maybe not a JS function here?

Reply:
Added the comment.

Reply:
> Could we instead remove `__throw_exception_with_stack_trace` and have `_Unwind_RaiseException` modify its behaviour based on `EXCEPTION_STACK_TRACES` in JS code?
> 
> Or is `_Unwind_RaiseException` maybe not a JS function here?

`_Unwind_RaiseException` is a function in libunwind: https://github.com/emscripten-core/emscripten/blob/36c194dfff84504434f8db4b8161071fa71f6c97/system/lib/libunwind/src/Unwind-wasm.c#L50-L56

We can move the differing behavior from libc++abi to libunwind, but given that libunwind is meant to be a low-level library and our `Unwind-wasm.c` is mostly a stub to call wasm instruction builtins, I feel libc++abi is a better place for calling out to JS. Also I like having the debug version of libc++abi for other reasons, such as users can see the `abort` error messages now, without which the errors could be very confusing and there was no way to build libc++abi in debug mode, even when `ASSERTIONS` was on.